manager/searcher/searcher.go

```diff
@@ -91,83 +97,113 @@ func New(pluginDir string) Searcher {
+		return nil, errors.New("empty scheduler clusters")
 	}
 
-	// If there are security domain conditions, match clusters of the same security domain.
-	// If the security domain condition does not exist, it will match all scheduler security domains.
-	// Then use clusters sets to score according to scopes.
-	var clusters []model.SchedulerCluster
-	securityDomain := conditions[ConditionSecurityDomain]
-	if securityDomain == "" {
-		logger.Infof("dfdaemon %s %s have empty security domain", client.HostName, client.Ip)
+	clusters := FilterSchedulerClusters(conditions, schedulerClusters)
+	if len(clusters) == 0 {
+		return nil, fmt.Errorf("security domain %s does not match any scheduler cluster", conditions[ConditionSecurityDomain])
 	}
 
-	for _, schedulerCluster := range schedulerClusters {
-		if len(schedulerCluster.Schedulers) > 0 {
-			if securityDomain == "" {
-				clusters = append(clusters, schedulerCluster)
-			} else {
-				for _, securityRule := range schedulerCluster.SecurityGroup.SecurityRules {
-					if strings.Compare(securityRule.Domain, securityDomain) == 0 {
-						clusters = append(clusters, schedulerCluster)
-					}
-				}
+	sort.Slice(
+		clusters,
+		func(i, j int) bool {
+			var si, sj Scopes
+			if err := mapstructure.Decode(clusters[i].Scopes, &si); err != nil {
+				logger.Errorf("cluster %s decode scopes failed: %v", clusters[i].Name, err)
+				return false
 			}
-		}
-	}
 
-	switch len(clusters) {
-	case 0:
-		// If the security domain does not match, there is no cluster available
-		return model.SchedulerCluster{}, fmt.Errorf("security domain %s does not match", securityDomain)
-	case 1:
-		// If only one cluster matches the security domain, return the cluster directly
-		return clusters[0], nil
-	default:
-		// If there are multiple clusters matching the security domain,
-		// select the schuelder cluster with a higher score
-		var maxScore float64 = 0
-		result := clusters[0]
-		for _, cluster := range clusters {
-			var scopes Scopes
-			if err := mapstructure.Decode(cluster.Scopes, &scopes); err != nil {
-				logger.Infof("cluster %s decode scopes failed: %v", cluster.Name, err)
-				// Scopes parse failed to skip this evaluation
-				continue
+			if err := mapstructure.Decode(clusters[j].Scopes, &sj); err != nil {
+				logger.Errorf("cluster %s decode scopes failed: %v", clusters[i].Name, err)
+				return false
 			}
 
-			score := Evaluate(conditions, scopes)
-			if score > maxScore {
-				maxScore = score
-				result = cluster
+			return Evaluate(conditions, si, clusters[i].SecurityGroup.SecurityRules) > Evaluate(conditions, sj, clusters[j].SecurityGroup.SecurityRules)
+		},
+	)
+
+	return clusters, nil
+}
+
+// Filter the scheduler clusters that dfdaemon can be used
+func FilterSchedulerClusters(conditions map[string]string, schedulerClusters []model.SchedulerCluster) []model.SchedulerCluster {
+	var clusters []model.SchedulerCluster
+	securityDomain := conditions[ConditionSecurityDomain]
```

Comment:
securityDomain,ok := conditions[ConditionSecurityDomain]

Reply:
```golang
m := map[string]string{"security_domain": ""}
v, ok := m["security_domain"]
```

The value of ok is true, but `security_domain` is empty.